scripts/lib/zulip_tools.py

```diff
@@ -106,12 +106,15 @@ def subprocess_text_output(args):
     # type: (Sequence[str]) -> str
     return subprocess.check_output(args, universal_newlines=True).strip()
 
+def get_zulip_uid() -> int:
+    return os.stat(get_deploy_root()).st_uid
+
 def su_to_zulip():
     # type: () -> None
-    pwent = pwd.getpwnam("zulip")
+    pwent = pwd.getpwuid(get_zulip_uid())
     os.setgid(pwent.pw_gid)
     os.setuid(pwent.pw_uid)
-    os.environ['HOME'] = os.path.abspath(os.path.join(DEPLOYMENTS_DIR, '..'))
+    os.environ['HOME'] = pwent.pw_dir
```

Comment:
So, I just thought of a problem with this -- when you're running the installer as `scripts/setup/install` as root, the root Zulip directory is owned by `root`, not `zulip`.  

Probably we should add a big docstring on top of `su_to_zulip` that this cannot be called from early production installer code that runs before `/home/zulip/deployments/current` exists.

Reply:
Maybe this?

    """Warning: su_to_zulip assumes that the zulip checkout is owned by                                                                
    the zulip user (or whatever normal user is running the Zulip                                                                       
    installation).  It should never be run from the installer or other                                                                 
    production contexts before /home/zulip/deployments/current is                                                                      
    created.                                                                                                                           
    """